### PR TITLE
Add ability to support branch coverage statistics ruby >= 2.5

### PR DESCRIPTION
**Pull request by som4ik**
Show statistics of branch coverage result.
branch1
all_files
You can put this merge on pending until they [merge](https://github.com/colszowka/simplecov/pull/692) `simplecov` gem branch that generates needed statistics. Anyway, these modifications are not harmful.

### CODE REVIEW

File: assets/stylesheets/screen.css.sass [hunk 1]
```diff
@@ -217,4 +221,9 @@ td
     &:nth-child(odd)
       background-color: #FBF0C0
     &:nth-child(even)
-      background-color: #FBFfCf
\ No newline at end of file
+      background-color: #FBFfCf
```
Comment (magneland) [inline assets/stylesheets/screen.css.sass]: How about using lowercase values for all colors in this file for consistency?

File: lib/simplecov-html.rb [hunk 2]
```diff
@@ -29,6 +30,20 @@ def output_message(result)
         "Coverage report generated for #{result.command_name} to #{output_path}. #{result.covered_lines} / #{result.total_lines} LOC (#{result.covered_percent.round(2)}%) covered."
       end
 
+      # Check if branchable results supported or not
+      # Try used here to escape exceptions
+      def branchable_result?
+        defined?(SimpleCov::branchable_report) ? SimpleCov::branchable_report : false
```
Comment (magneland) [inline lib/simplecov-html.rb]: It's not clear to me what `SimpleCov::branchable_report` refers to? Should it be `SimpleCov.branchable_report`?

File: lib/simplecov-html.rb [hunk 3]
```diff
@@ -29,6 +30,20 @@ def output_message(result)
...
+      end
+
+      def line_status?(source_file, line)
+        if branchable_result? && source_file.line_with_missed_branch?(line.number)
+          "missed-branch"
```
Comment (magneland) [inline lib/simplecov-html.rb]: Not sure it matters, but this could be made into a constant / frozen string.

File: views/source_file.erb [hunk 4]
```diff
@@ -1,22 +1,57 @@
...
+    <h4>
+      <span class="<%= coverage_css_class(source_file.covered_percent) %>">
+        <%= source_file.covered_percent.round(2).to_s %> %
+      </span>
+      lines covered
+    </h4>
+
+    <% if branchable_result? %>
+      <h4>
+        <span class="<%= coverage_css_class(source_file.branches_coverage_precent) %>">
```
Comment (magneland) [inline views/source_file.erb]: Typo: percent

**Comment by PragTob** [comment]
Ugh all the CI failing is my/our fault... need to fix that up. Sorry.

**Comment by som4ik** [comment]
@PragTob I updated the branch, hope it will fix CI, else I will try to have a look this evening!

**Comment by som4ik** [comment]
> Ugh all the CI failing is my/our fault... need to fix that up. Sorry.
`.travis.yml` updated according to the release https://github.com/colszowka/simplecov/releases/tag/v0.17.0

**Comment by PragTob** [comment]
Ah wanted to fix that up on a separate branch, small changes, fix it for everyone and not tie it to this branch and such... but thank you still, I'll do further fixing after this landed which will hopefully be soon :D

**Comment by amatsuda** [comment]
🎉

**Pull request merged**